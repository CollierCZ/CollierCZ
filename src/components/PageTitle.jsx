import React from "react";
import { graphql, StaticQuery } from "gatsby";
import { css } from '@emotion/core'

class PageTitle extends React.Component {
  render() {
    const { text } = this.props;
    return (
      <StaticQuery
        query={graphql`
          query {
            kontentItemHome{
              elements {
                title {
                  value
                }
              }
            }
          }
        `}
        render = {data => {
          return (
            <h1
              css={css`
                margin: 10px 0;
                font-size: 5rem;
                letter-spacing: -1px;
                font-weight: 700;
                font-family: "Open Sans", sans-serif;
                color: #ffffff;
                text-shadow: 1px 1px #000;
                @media only screen and (max-width: 900px) {
                  font-size: 4rem;
                  letter-spacing: -1px;
                }
                @media only screen and (max-width: 500px) {
                  font-size: 3rem;
                }
              `}
            >{text ? text : data.kontentItemHome.elements.title.value}</h1>
          )
        }}
      />
    )
  }
}

export default PageTitle;
