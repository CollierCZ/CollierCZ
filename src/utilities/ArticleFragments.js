import { graphql } from "gatsby"

export const ArticleListFragment = graphql`
  fragment ArticleListFragment on KontentItemArticle {
    fields {
      slug
      tags
      date
    }
    elements {
      metadata__description {
        value
      }
      metadata__keywords {
        value
      }
      categories {
        value {
          name
        }
      }
      title {
        value
      }
      teaser {
        value {
          url
        }
      }            
    }
  }
`