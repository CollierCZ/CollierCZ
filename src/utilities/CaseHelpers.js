const slugify = (text) => {
  return text
    .toString()
    .toLowerCase()
    .trim()
    .replace(/\s+/g, '-')       // Replace spaces with -
    .replace(/[^\w-]+/g, '')     // Remove all non-word chars
}

const titleCase = (text) => {
  return(
    text.replace(/\w\S*/g, function(txt) {
      return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
    })
  )
}

module.exports =
{
  slugify: slugify,
  titleCase: titleCase
}