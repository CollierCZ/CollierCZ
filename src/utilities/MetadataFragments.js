import { graphql } from "gatsby"

export const MetadataFragment = graphql `
  fragment MetadataFragment on kontent_item_home {
    elements {
      title {
        value
      }
      splash_image {
        value {
          url
        }
      }
      blog_logo {
        value {
          url
        }
      }
      metadata__description {
        value
      }
      socialmedia {
        value
      }
      base_url {
        value
      }
    }
  }
`

export const SocialMediaFragment = graphql `
  fragment SocialMediaFragment on kontent_item_home {
    elements {
      socialmedia {
        value
      }
    }
  }
`