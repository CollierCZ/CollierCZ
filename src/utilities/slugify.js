let slugify = (text) => {
    return text
        .toString()
        .toLowerCase()
        .trim()
        .replace(/\s+/g, '-')           // Replace spaces with -
        .replace(/[^\w-]+/g, '')       // Remove all non-word chars
}


module.exports.slugify = slugify