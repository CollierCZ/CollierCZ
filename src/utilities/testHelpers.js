import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { navigate } from "gatsby";

export const simpleRenderer = component => {
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
};
export const clickTextLink = component => {
  const mountedComponent = mount(component);
  mountedComponent
    .find("a")
    .first()
    .simulate("click");
  expect(navigate).toHaveBeenCalled();
};
