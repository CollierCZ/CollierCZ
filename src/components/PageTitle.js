import React from "react";
import styled from "styled-components";
import { titleCase } from "../utilities/CaseHelpers";

import Heading from "@kiwicom/orbit-components/lib/Heading";
import { useSiteMetadata } from "../hooks/use-metadata";

const StyledHeading = styled("div")`
  text-shadow: 1px 1px #000;
`;

export const PurePageTitle = ({ text, siteMetadata }) => {
  text = text ? titleCase(text) : null;
  return (
    <StyledHeading>
      <Heading element="h1" type="display" inverted>
        {text ? text : siteMetadata.elements.title.value}
      </Heading>
    </StyledHeading>
  );
};

export const PageTitle = ({ text }) => {
  const siteMetadata = useSiteMetadata();
  return <PurePageTitle text={text} siteMetadata={siteMetadata} />;
};

export default PageTitle;
