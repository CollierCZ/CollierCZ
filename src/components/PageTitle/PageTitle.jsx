import React from "react";
import { graphql, StaticQuery } from "gatsby";
import "./PageTitle.css";

class PageTitle extends React.Component {
  render() {
    const { text } = this.props;
    return (
      <StaticQuery
        query={graphql`
          query {
            kenticoCloudItemHome{
              elements {
                title {
                  value
                }
              }
            }
          }
        `}
        render = {data => {
          return (
            <h1 className="page-title">{text ? text : data.kenticoCloudItemHome.elements.title.value}</h1>
          )
        }}
      />
    )
  }
}

export default PageTitle;
