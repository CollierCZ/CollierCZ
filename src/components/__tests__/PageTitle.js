import React from "react";
import { PurePageTitle as PageTitle } from "../PageTitle";
import { simpleRenderer } from "../../utilities/testHelpers";

const siteMetadata = {
  elements: {
  title: {
  value: 'A title'
  }
}
}

describe("PageTitle", () => {
  it("renders correctly with title", () => {
    simpleRenderer(<PageTitle text="work elsewhere" />);
  });
  it("renders correctly with nothing", () => {
    simpleRenderer(<PageTitle siteMetadata={siteMetadata} />);
  });
});
