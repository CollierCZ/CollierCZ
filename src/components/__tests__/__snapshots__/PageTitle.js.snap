// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle renders correctly with nothing 1`] = `
.c1 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 40px;
  font-weight: 700;
  color: #FFFFFF;
  line-height: 44px;
  margin: 0;
}

.c0 {
  text-shadow: 1px 1px #000;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    A title
  </div>
</div>
`;

exports[`PageTitle renders correctly with title 1`] = `
.c1 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 40px;
  font-weight: 700;
  color: #FFFFFF;
  line-height: 44px;
  margin: 0;
}

.c0 {
  text-shadow: 1px 1px #000;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Work Elsewhere
  </div>
</div>
`;
